packages/analytics/amplify_analytics_pinpoint_dart/test/common/endpoint_global_fields_storage.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:convert';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/endpoint_client/endpoint_store_keys.dart';
+import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
+
+/// Retrieve attributes map from [SecureStorageInterface]
+Future<Map<String, String>> getStoredAttributes(
```

Comment:
Why are you using these helpers instead of `EndpointGlobalFieldsManager`?

These make your tests very brittle because their tied to the implementation - if your implementation changes, these tests need to change. If you used `EndpoingGlobalFieldsManager`, the implementation can change and your tests then work to verify your changes are accurate.

Reply:
Good point.  Exposed internal getAttributes and getMetrics from `EndpointGlobalFieldsManager` to be used in tests.